inc/StyleUtils.php

```diff
@@ -4,101 +4,161 @@
+                            foreach($config['stylesheets'] as $file => $mode) {
+
+                                // validate and include style files
+                                $stylesheets = array_merge($stylesheets, $this->getValidatedStyles($stylesheets, $file, $mode, $incbase, $webbase));
+                                $combined['stylesheets'] = is_array($combined['stylesheets']) ? array_merge($combined['stylesheets'], $stylesheets) : $stylesheets;
+                            }
+                        }
+
+                        if (is_array($config['replacements'])) {
+                            $replacements = array_replace($replacements, $this->cssFixreplacementurls($config['replacements'], $webbase));
+                            $combined['replacements'] = array_merge($replacements, $config['replacements']);
+                        }
+                    }
+                }
             }
+
         }
+        return $combined;
+    }
 
-        // allow replacement overwrites in preview mode
-        if($preview) {
-            $webbase = DOKU_BASE;
-            $ini     = $conf['cachedir'].'/preview.ini';
-            if(file_exists($ini)) {
-                $data = parse_ini_file($ini, true);
-                // replacements
-                if(is_array($data['replacements'])) {
-                    $replacements = array_merge($replacements, $this->cssFixreplacementurls($data['replacements'], $webbase));
+    /**
+     * Checks if configured style files exist and, if necessary, adjusts file extensions in config
+     *
+     * @param array $stylesheets
+     * @param string $file
+     * @param string $mode
+     * @param string $incbase
+     * @param string $webbase
+     * @return mixed
+     */
+    protected function getValidatedStyles($stylesheets, $file, $mode, $incbase, $webbase) {
+        global $conf;
+        if (!file_exists($incbase . $file)) {
+            list($extension, $basename) = array_map('strrev', explode('.', strrev($file), 2));
+            $newExtension = $extension === 'css' ? 'less' : 'css';
+            if (file_exists($incbase . $basename . '.' . $newExtension)) {
+                $stylesheets[$mode][$incbase . $basename . '.' . $newExtension] = $webbase;
+                if ($conf['allowdebug']) {
+                    msg("Stylesheet $file not found, using $basename.$newExtension instead. Please contact developer of \"{$conf['template']}\" template.", 2);
                 }
             }
         }
-
-        return array(
-            'stylesheets' => $stylesheets,
-            'replacements' => $replacements
-        );
+        $stylesheets[$mode][$incbase . $file] = $webbase;
+        return $stylesheets;
     }
 
+    /**
+     * Returns the server path for the given level/group in config cascade
+     *
+     * @param string $config_group
+     * @return string
+     */
+    protected function getServerPath($config_group) {
```

Comment:
I'm not happy with the method name. ServerPath sounds too much like a file system path, but this is the URI path. How about `getBaseUrl()` or `getWebBase()` or something similar?

Some explanation what this does might be useful in the doc block. Like "style resources are relative to the template directory for the main (default) styles but relative to DOKU_BASE for everything else"